### PR TITLE
Refactor admin panel (IN PROGRESS)

### PR DESCRIPTION
**Pull request by mateossh**
I wanted to refactor entire admin panel before Hacktoberfest ends. Time has almost run out and I'm nowhere near finish line. 

At the beginning I started writing new components (because old admin panel had 4 iirc). I couldn't test my new code, because I used async and project didn't build. I didn't care about it and I kept rewriting. In the meantime I noticed old code wasn't following ESLint rules so I also updated them a bit to make code more readable. When new components were made I fixed issue with building project.

Things which has to be done before merging to master:
- [ ] Main issue: because project uses SSR there's issues with URL links to API when project is built.
- [ ] New components has no styling
- [ ] Probably there's a lot more ...

I don't have any good solution to fix first issue.

I encourage you to help me and make this project much better.

### CODE REVIEW

**Comment by vkuznecovas** [comment]
Super excited about the submission. Will try and get my hands on this ASAP and give you feedback

File: admin/src/components/AllComments/index.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
+import { h, Component } from 'preact';
+import Thread from '../Thread';
+import Comment from '../Comment';
+
+export default class AllComments extends Component {
+  render() {
+    // NOTE: this looks like shit. gonna rewrite it to make it more readible
```
Comment (vkuznecovas) [inline admin/src/components/AllComments/index.js]: Swearing is never good in code. Please provider a proper todo here.

File: admin/src/components/AllComments/index.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+    const threads = this.props.threads.sort((a, b) => {
+      a = new Date(a.CreatedAt);
+      b = new Date(b.CreatedAt);
+      return a > b ? -1 : a < b ? 1 : 0;
```
Comment (vkuznecovas) [inline admin/src/components/AllComments/index.js]: nested ternaries are rather nasty to read, can we refactor this bit?

File: admin/src/components/AllComments/index.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
+import { h, Component } from 'preact';
```
Comment (vkuznecovas) [inline admin/src/components/AllComments/index.js]: the h is not used in code, can we get rid of it?

File: admin/src/components/Comment/index.js [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+    } catch (err) {
+      console.log('Error while updating comment', err);
+    }
+  }
+
+  async confirmComment() {
+    // NOTE: dunno what should I think about this
+    try {
+      const data = {
+        // CommentId: this.props.id,
```
Comment (vkuznecovas) [inline admin/src/components/Comment/index.js]: there's no specific confirm endpoint on the API. We're using the update endpoint to confirm comments, so these should be uncommented in order not to lose data.

File: admin/src/components/Comment/index.js [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+        // Body: this.props.body,
+        // Author: this.props.author,
+        Confirmed: true,
```
Comment (vkuznecovas) [inline admin/src/components/Comment/index.js]: do we also need to update the comment state. We don't need to show the button if the comment is already confirmed.

File: admin/src/components/DeletedComments/index.js [hunk 6]
```diff
@@ -0,0 +1,29 @@
+import { h, Component } from 'preact';
+import Thread from '../Thread';
+import Comment from '../Comment';
+
+export default class DeletedComments extends Component {
+  render() {
+    // NOTE: this looks like shit. gonna rewrite it to make it more readable
```
Comment (vkuznecovas) [inline admin/src/components/DeletedComments/index.js]: Swearing is never good in code. Please provider a proper todo here.

File: admin/src/components/Menu/index.js [hunk 7]
```diff
@@ -0,0 +1,14 @@
+import { h, Component } from 'preact';
+import { Link } from 'preact-router/match';
+
+export default class Menu extends Component {
+  render() {
+    return (
+      <nav>
+        <Link href="/">All comments</Link>
```
Comment (vkuznecovas) [inline admin/src/components/Menu/index.js]: it's not really possible to do a proper link here, as the files are served as static content. So we'll need to think about this.

File: admin/src/components/PendingComments/index.js [hunk 8]
```diff
@@ -0,0 +1,29 @@
+import { h, Component } from 'preact';
+import Thread from '../Thread';
+import Comment from '../Comment';
+
+export default class PendingComments extends Component {
+  render() {
+    // NOTE: this looks like shit. gonna rewrite it to make it more readable
```
Comment (vkuznecovas) [inline admin/src/components/PendingComments/index.js]: Swearing is never good in code. Please provider a proper todo here.

File: admin/src/components/PendingComments/index.js [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+
+    const threads = this.props.threads.sort((a, b) => {
+      a = new Date(a.CreatedAt);
+      b = new Date(b.CreatedAt);
+      return a > b ? -1 : a < b ? 1 : 0;
```
Comment (vkuznecovas) [inline admin/src/components/PendingComments/index.js]: nested ternary :/

File: admin/src/components/Menu/index.js (hunk 7 above)
Comment (mateossh) [reply admin/src/components/Menu/index.js]: Those links redirects to components defined in router. Everything is good here

**Comment by mateossh** [comment]
I give up with this pull request. 

There's too much new things in this project for me: Preact, SSR, Golang on server-side. My brain can't "decrypt" how existing admin panel works or debug why API endpoints sends 401 with Postman... despite this is so simple project.

**Pull request closed without merging**